src/libopal.c

```diff
@@ -2852,8 +2852,40 @@ gen_obj (char *asm_fn, char *obj_fn)
   /// Assert object file name is not null
   assert(obj_fn);
 
-  // TODO
+  /// Create the NASM call with -g, -f and -o flags
+  logger(DEBUG, "Calling NASM to assemble object.");
+  char NASM_cmd[1024] = { 0 };
+  sprintf(NASM_cmd, "NASM -g -f elf64 -o %s %s", obj_fn, asm_fn);
+
+  /// Check if asm_fn can be read
+  logger(DEBUG, "access (%s, R_OK)", asm_fn);
+  errno = EXIT_SUCCESS;
+  if (access(asm_fn, R_OK) == EXIT_SUCCESS)
+  {
+      _PASS;
+
+      /// Check if obj_fn can be written to
+      logger(DEBUG, "access (%s, W_OK)", obj_fn);
+      errno = EXIT_SUCCESS;
+      if (access(obj_fn,  W_OK) == EXIT_SUCCESS)
```

Comment:
You should check if obj_fn can be **read** after `nasm` is run.

Right now, it will fail at this point, since the `obj_fn` doesn't exist yet.